### PR TITLE
Merge ANCM into IISIntegration, Round 2

### PR DESCRIPTION
**Pull request by jkotalik**
Fixes:
- Removes ANCM tests for now. There are cyclic dependencies and odd package versions that need to be resolved. I will fix this in the next iteration.
- Removes any .All references
- Fixes any package publishing issues.

I see this passing now through most of universe locally. I ran into a test failure in MvcPrecompilation, will rerun now to see if it is flaky.

### CODE REVIEW

File: build/repo.targets [hunk 1]
```diff
@@ -16,11 +20,11 @@
       Condition="'$(VisualStudioMSBuildx86Path)' != ''" />
   </Target>
 
-  <Target Name="PackageProjects">
+  <Target Name="PackageNativeProjects">
     <MSBuild
       Projects="$(MSBuildThisFileDirectory)../nuget/AspNetCore.csproj"
       Targets="Restore;Pack"
-      Properties="PackageVersion=$(PackageVersion);PackageOutputPath=$(BuildDir)" />
+      Properties="PackageVersion=$(PackageVersion);PackageOutputPath=$(BuildDir);BuildNumber=$(BuildNumber)" />
```
Comment (natemcmaster) [inline build/repo.targets]: Add one more variable. `IsFinalBuild=$(IsFinalBuild)`

**Comment by natemcmaster** [comment]
Can you also cherry-pick this change? https://github.com/aspnet/AspNetCoreModule/commit/39c24813a6d3310d79ea541ab44597b92f9e47f2

**Comment by jkotalik** [comment]
I'll cherry pick all changes in ANCM. I'll wait to merge https://github.com/aspnet/AspNetCoreModule/pull/274 until this is in, and then make a new PR here.

**Comment by jkotalik** [comment]
Merged in.

**Pull request closed without merging**